icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java

```diff
@@ -0,0 +1,366 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+
+package com.ibm.icu.dev.test.impl;
+
+import com.ibm.icu.dev.test.TestUtil;
+import com.ibm.icu.impl.Pair;
+import com.ibm.icu.impl.units.ComplexUnitsConverter;
+import com.ibm.icu.impl.units.ConversionRates;
+import com.ibm.icu.impl.units.MeasureUnitImpl;
+import com.ibm.icu.impl.units.UnitConverter;
+import com.ibm.icu.impl.units.UnitsRouter;
+import com.ibm.icu.util.Measure;
+import com.ibm.icu.util.MeasureUnit;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+
+public class UnitsTest {
```

Comment:
Optional: consider sorting the tests in the same order as in units_test.cpp? Otherwise I would sort them myself next time I touch these files (and then "git blame" gets updated).

I noticed this PR does not have:

    void testUnitConstantFreshness();
    void testSiPrefixes();
    void testMass();
    void testArea();

I recognise the last three can be folded together with the temperature test:

    public void testConverterForTemperature() {

Do you think we should sync up the tests later, rather than now? If so, as part of which ticket?

Reply:
Taken care of in https://github.com/unicode-org/icu/pull/1421